res/controllers/Traktor Kontrol S4 MK3.hid.xml

```diff
@@ -9,6 +9,615 @@
+                    <value label="Album">3</value>
+                    <value label="Albumartist">4</value>
+                    <value label="Year">5</value>
+                    <value label="Genre">6</value>
+                    <value label="Composer">7</value>
+                    <value label="Grouping">8</value>
+                    <value label="Tracknumber">9</value>
+                    <value label="Filetype">10</value>
+                    <value label="Native Location">11</value>
+                    <value label="Comment">12</value>
+                    <value label="Duration">13</value>
+                    <value label="Bitrate">14</value>
+                    <value label="BPM">15</value>
+                    <value label="ReplayGain">16</value>
+                    <value label="Datetime Added">17</value>
+                    <value label="Times Played">18</value>
+                    <value label="Rating">19</value>
+                    <value label="Key" default="true">20</value>
+                    <description>Fourth column when iterating over sortable library columns using VIEW+Library Encoder press</description>
+                </option>
+                <option
+                    variable="librarySortableColumns5Value"
+                    type="enum"
+                    label="Fifth column">
+                    <value label="---">0</value>
+                    <value label="Artist">1</value>
+                    <value label="Title">2</value>
+                    <value label="Album">3</value>
+                    <value label="Albumartist">4</value>
+                    <value label="Year">5</value>
+                    <value label="Genre">6</value>
+                    <value label="Composer">7</value>
+                    <value label="Grouping">8</value>
+                    <value label="Tracknumber">9</value>
+                    <value label="Filetype">10</value>
+                    <value label="Native Location">11</value>
+                    <value label="Comment">12</value>
+                    <value label="Duration">13</value>
+                    <value label="Bitrate">14</value>
+                    <value label="BPM">15</value>
+                    <value label="ReplayGain">16</value>
+                    <value label="Datetime Added" default="true">17</value>
+                    <value label="Times Played">18</value>
+                    <value label="Rating">19</value>
+                    <value label="Key">20</value>
+                    <description>Fifth column when iterating over sortable library columns using VIEW+Library Encoder press</description>
+                </option>
+                <option
+                    variable="librarySortableColumns6Value"
+                    type="enum"
+                    label="Sixth column">
+                    <value label="---" default="true">0</value>
+                    <value label="Artist">1</value>
```

Comment:
will we have to keep these values in sync with code? Can we add a test that breaks if those change?

Reply:
It could be a gross gtest in c++ that loads the xml and compares it to the C++ code.  or maybe a better option is we could make this file somehow templated (qtxml?) and generated as part of the build process.  

The library changes not very often, but often enough. Without an automated way of detecting the change I'm afraid we will never know that we broke this config.

Reply:
I went with Option 1 - let me know if that's good enough.